### DISCUSSION TITLE
Question about headless testing

### DISCUSSION THREAD

**Discussion by rabbitism** [AvaloniaUI/Avalonia#15279] [Q&A] [answered]
I setup a test project according to doc https://docs.avaloniaui.net/docs/concepts/headless/headless-xunit.

When I change a control property that affects measure and render, it does not take effect.
I have to manually call `Dispatcher.UJIThread.RunJobs()`.

Is this the expected usage?

**Comment by timunie** [comment]
@rabbitism would be nice to see a sample usage to judge it better

**Reply by rabbitism** [reply]
OK, I need some time to setup a sample repo~

**Comment by maxkatz6** [comment, marked as answer]
Yeah, expected - https://docs.avaloniaui.net/docs/concepts/headless/#simulating-user-delay

**Reply by rabbitism** [reply]
thanks!
